x-pack/plugins/ml/public/application/components/field_stats_flyout/field_stats_flyout_provider.tsx

```diff
@@ -27,6 +48,77 @@ export const FieldStatsFlyoutProvider: FC<{
     () => setFieldStatsIsFlyoutVisible(!isFieldStatsFlyoutVisible),
     [isFieldStatsFlyoutVisible]
   );
+  const [manager] = useState(new PopulatedFieldsCacheManager());
+  const [populatedFields, setPopulatedFields] = useState<Set<string> | undefined>();
+  const abortController = useRef(new AbortController());
+
+  useEffect(
+    function fetchSampleDocsEffect() {
+      if (disablePopulatedFields) return;
+
+      const queryAndRunTimeMappings = getMergedSampleDocsForPopulatedFieldsQuery({
+        searchQuery: dslQuery,
+        runtimeFields: dataView.getRuntimeMappings(),
+        datetimeField: dataView.getTimeField()?.name,
+        timeRange: timeRangeMs,
+      });
+      const indexPattern = dataView.getIndexPattern();
+      const esSearchRequestParams = {
+        index: indexPattern,
+        body: {
+          fields: ['*'],
+          _source: false,
+          ...queryAndRunTimeMappings,
+          size: 1000,
+        },
+      };
+      const cacheKey = stringHash(JSON.stringify(esSearchRequestParams)).toString();
+
+      const fetchSampleDocuments = async function () {
+        try {
+          const resp = await lastValueFrom(
+            search.search(
+              {
+                params: esSearchRequestParams,
+              },
+              { abortSignal: abortController.current.signal }
+            )
+          );
+
+          const docs = resp.rawResponse.hits.hits.map((d) => getProcessedFields(d.fields ?? {}));
+
+          // Get all field names for each returned doc and flatten it
+          // to a list of unique field names used across all docs.
+          const fieldsWithData = new Set(docs.map(Object.keys).flat(1));
+          manager.set(cacheKey, fieldsWithData);
+          setPopulatedFields(fieldsWithData);
```

Comment:
there is a possibility of setting a state for unmounted component here

Reply:
Updated here [`a01491e` (#163371)](https://github.com/elastic/kibana/pull/163371/commits/a01491ea644fed0fd9792523a2beda47e900d587)